storage/src/mmr/verification.rs

```diff
@@ -234,15 +242,19 @@ impl<H: CHasher> Proof<H> {
                 siblings.sort_by(|a, b| b.0.cmp(&a.0));
             }
         }
-        hashes.extend(
-            siblings
-                .iter()
-                .map(|(_, pos)| mmr.get_node(*pos).unwrap().clone()),
-        );
-        Proof {
-            size: mmr.size(),
+        hash_futures.extend(siblings.iter().map(|(_, pos)| mmr.get_node(*pos)));
+        let hash_results = join_all(hash_futures).await;
```

Comment:
You can use `try_join_all` to wait for everything or stop at the first error: https://docs.rs/futures/latest/futures/future/fn.try_join_all.html

Reply:
NICE. done.